File: app-android/feature/ai-chat/src/main/java/com/aidatainsight/android/feature/aichat/domain/AIChatRepository.kt
package com.aidatainsight.android.feature.aichat.domain

import com.aidatainsight.android.core.model.contract.FunctionArguments
import com.aidatainsight.android.core.model.contract.FunctionModel
import com.aidatainsight.android.core.model.contract.FunctionName
import com.aidatainsight.android.core.model.contract.HistoryChartDetail
import com.aidatainsight.android.core.model.contract.HistoryRecord
import com.aidatainsight.android.core.model.contract.TemplateQuestionSet
import kotlinx.coroutines.flow.Flow

interface AIChatRepository {
    suspend fun loadTemplate(): TemplateQuestionSet
    suspend fun loadHistoryDetail(historyId: Int): HistoryRecord
    suspend fun sendFunctionMessage(text: String, historyId: Int?): FunctionModel
    suspend fun loadChartData(
        name: FunctionName,
        historyId: Int,
        arguments: FunctionArguments,
    ): HistoryChartDetail
    suspend fun sendLikeFeedback(historyDetailId: Int, like: String)
    fun streamMessage(text: String): Flow<String>
}

File: Packages/module-ai/Sources/ModuleAI/Domain/AIChat/AIChatRepository.swift
//
//  AIChatRepository.swift
//  ModuleAI
//
//  Created by Codex on 2026/1/1.
//

import Foundation

protocol AIChatRepository {
    func loadTemplate() async throws -> TemplateModel
    func loadHistoryDetail(_ historyId: Int) async throws -> RecordModel
    func sendFunctionMessage(_ text: String, historyId: Int?) async throws -> FunctionModel
    func loadChartData(name: FunctionName, historyId: Int, arguments: FunctionArguments) async throws -> HistoryDetailModel
    func sendLikeFeedback(historyDetailId: Int, like: String) async throws
    func streamMessage(_ text: String) -> AsyncThrowingStream<String, Error>
}
